machine.go

```diff
@@ -755,13 +761,14 @@ func (m *Machine) createNetworkInterface(ctx context.Context, iface NetworkInter
 	}
 
 	m.logger.Debugf("createNetworkInterface returned for %s", iface.StaticConfiguration.HostDevName)
+	m.networkInterfaces[ifaceID] = ifaceCfg
 	return err
 }
 
 // UpdateGuestNetworkInterfaceRateLimit modifies the specified network interface's rate limits
 func (m *Machine) UpdateGuestNetworkInterfaceRateLimit(ctx context.Context, ifaceID string, rateLimiters RateLimiterSet, opts ...PatchGuestNetworkInterfaceByIDOpt) error {
```

Comment:
The opts will be ignored on the PUT path. How about removing `opts`? Nobody is using the method, at least on GitHub.

https://github.com/search?q=UpdateGuestNetworkInterfaceRateLimit&type=Code

Reply:
@xibz @IRCody What do you think? I think removing that (while nobody is using, at least on GitHub) is okay.

Reply:
If we remove it are we forcing breaking API changes in the future if FC adds an option here?

Reply:
It depends how we model their changes on the Go SDK. If Firecracker extends NetworkInterface / PartialNetworkInterface and adding opts makes sense from our perspective, then yes.

Another option is supporting only PatchGuestNetworkInterface. We need to update our tests and we might break other clients (technically Firecracker's change breaks the clients), but it is simpler and we are making less arbitrary decisions.

Reply:
You mean only supporting PatchGuestNetworkInterface in UpdateGuestNetworkInterfaceRateLimit?

Reply:
Yes. And doing the same on UpdateGuestDrive()

Reply:
I've updated the implementation to only support PatchGuestNetworkInterface (post-boot configuration).